presentation-compiler/src/main/dotty/tools/pc/printer/ShortenedTypePrinter.scala

```diff
@@ -0,0 +1,546 @@
+      case tp if tp.isError => super.toText(indexedCtx.ctx.definitions.AnyType)
+      case _ => super.toText(tp)
+
+  override def toTextSingleton(tp: SingletonType): Text =
+    tp match
+      case ConstantType(const) => toText(const)
+      case _ => toTextRef(tp) ~ ".type"
+
+  def tpe(tpe: Type): String = toText(tpe).mkString(defaultWidth, false)
+
+  def hoverSymbol(sym: Symbol, info: Type)(using Context): String =
+    val typeSymbol = info.typeSymbol
+
+    def shortTypeString: String = tpe(info)
+
+    def ownerTypeString: String =
+      typeSymbol.owner.fullNameBackticked
+
+    def name: String = nameString(sym)
+
+    sym match
+      case p if p.is(Flags.Package) =>
+        s"package ${p.fullNameBackticked}"
+      case c if c.is(Flags.EnumVal) =>
+        s"case $name: $shortTypeString"
+      // enum
+      case e if e.is(Flags.Enum) || sym.companionClass.is(Flags.Enum) =>
+        s"enum $name: $ownerTypeString"
+      /* Type cannot be shown on the right since it is already a type
+       * let's instead use that space to show the full path.
+       */
+      case o if typeSymbol.is(Flags.Module) => // enum
+        s"${keyString(o)} $name: $ownerTypeString"
+      case m if m.is(Flags.Method) =>
+        defaultMethodSignature(m, info)
+      case _ =>
+        val implicitKeyword =
+          if sym.is(Flags.Implicit) then List("implicit") else Nil
+        val finalKeyword = if sym.is(Flags.Final) then List("final") else Nil
+        val keyOrEmpty = keyString(sym)
+        val keyword =
+          if keyOrEmpty.iterator.nonEmpty then List(keyOrEmpty) else Nil
+        (implicitKeyword ::: finalKeyword ::: keyword ::: (s"$name:" :: shortTypeString :: Nil))
+          .mkString(" ")
+    end match
+  end hoverSymbol
+
+  def completionSymbol(sym: Symbol): String =
+    val info = sym.info.widenTermRefExpr
+    val typeSymbol = info.typeSymbol
+
+    if sym.is(Flags.Package) || sym.isClass then " " + fullNameString(sym.owner)
+    else if sym.is(Flags.Module) || typeSymbol.is(Flags.Module) then
+      if typeSymbol != NoSymbol then " " + fullNameString(typeSymbol.owner)
+      else " " + fullNameString(sym.owner)
+    else if sym.is(Flags.Method) then
+      defaultMethodSignature(sym, info, onlyMethodParams = true)
+    else tpe(info)
+
+  /**
+   * Compute method signature for the given (method) symbol.
+   *
+   * @return shortend name for types or the type for terms
+   *         e.g. "[A: Ordering](a: A, b: B): collection.mutable.Map[A, B]"
+   *              ": collection.mutable.Map[A, B]" for no-arg method
+   */
+  def defaultMethodSignature(
+      gsym: Symbol,
+      gtpe: Type,
+      onlyMethodParams: Boolean = false,
+      additionalMods: List[String] = Nil
+  ): String =
+    val namess = gtpe.paramNamess
+    val infoss = gtpe.paramInfoss
+    val nameToInfo: Map[Name, Type] = (for
+      pairs <- namess.zip(infoss).map { (names, infos) => names.zip(infos) }
+      pair <- pairs
+    yield pair).toMap
```

Comment:
Looks like this could be written more simply as 
```suggestion
    val nameToInfo: Map[Name, Type] =
      namess.flatten.lazyZip(infoss.flatten).toMap
```